### DISCUSSION TITLE
healthchecks UWSGI

### DISCUSSION THREAD

**Discussion by assistenza-studydata** [healthchecks/healthchecks#496] [Q&A]
i followed the instructions for putting into production but i am stuck on gunicorn or uwsgi.
I am a good programmer but I have never used python or django and I feel like a 3 year old.
i used this:
uwsgi --http 10.142.0.2:8000 --module hc/wsgi.py

but I get this
ModuleNotFoundError: No module named 'hc / wsgi'

i can't find the correct wsgi file, can you help me with the correct command? thank you

**Comment by cuu508** [comment]
uwsgi is a bit of a beast with tons of options and spotty documentation. It can be a pain to get it working for the first time, or when enabling a feature you have not used before. It's great once you get it right though :-)

In your example, try a different syntax for the --module option: `--module hc.wsgi:application`. 

When I tried this with uwsgi I had already installed system-wide, I had to do a couple extra bits. First, it said `--module` is not a recognized option. I had to also add `--plugin python3` to fix that. Second, I had the dependencies installed in a virtualenv, and it could not find it. The error message was "No module named 'django'". I added a `--virtualenv /path/to/my/venv` to fix that. This command worked for me:

    uwsgi --master --http-socket :8000 --plugin python3 --module hc.wsgi:application --virtualenv /path/to/my/venv

As you start adding options, it can be easier to put the configuration in a .ini file, and run it like so: `uwsgi uwsgi.ini`. [Here's an example ini file](https://github.com/healthchecks/healthchecks/blob/master/docker/uwsgi.ini). It assumes the project code lives at `/opt/healthchecks` and all dependencies are installed globally (not in a virtualenv).
